templates/common/baremetal/units/nodeip-configuration.service.yaml

```diff
@@ -13,14 +13,22 @@ contents: |
   [Service]
   # Need oneshot to delay kubelet
   Type=oneshot
-  ExecStart=/usr/bin/podman run --rm \
+  # Would prefer to do Restart=on-failure instead of this bash retry loop, but
+  # the version of systemd we have right now doesn't support it. It should be
+  # available in systemd v244 and higher.
+  ExecStart=/bin/bash -c " \
+    until \
+    /usr/bin/podman run --rm \
     --authfile /var/lib/kubelet/config.json \
     --net=host \
     --volume /etc/systemd/system:/etc/systemd/system:z \
     {{ .Images.baremetalRuntimeCfgImage }} \
     node-ip \
     set --retry-on-failure \
-    {{.Infra.Status.PlatformStatus.BareMetal.APIServerInternalIP }}
+    {{.Infra.Status.PlatformStatus.BareMetal.APIServerInternalIP }}; \
+    do \
+    sleep 5; \
```

Comment:
The service will be stuck in 'Active' state but the user (or whoever gonna troubleshoot) will have no idea that it's in a loop. Maybe to add log at least?
Another option is do podman pull in a loop with ExecStartPre phase - it will be more clear to the system - although there might be issues with failures, phase timeout and other systemd affects for oneshot services.

Reply:
Podman should be outputting an error message when it fails already. That's going to contain the information you need to debug the problem. I'm not sure adding a generic message in the service would gain us much.